src/reader/mod.rs

```diff
@@ -1630,33 +1654,13 @@ mod test {
                 use crate::reader::Reader;
                 use pretty_assertions::assert_eq;
 
-                /// When `encoding` feature is enabled, encoding should be detected
-                /// from BOM (UTF-8) and BOM should be stripped.
-                ///
-                /// When `encoding` feature is disabled, UTF-8 is assumed and BOM
-                /// character should be stripped for consistency
-                #[$test]
-                $($async)? fn bom_from_reader() {
-                    let mut reader = Reader::from_reader("\u{feff}\u{feff}".as_bytes());
```

Comment:
`Reader::from_reader()` isn't going to be implemented for `Reader<&[u8]>`, but the tests in this macro get implemented for both.

Reply:
To keep this test we would need to either specialize it fo each reader variant, or keep it as a unit-level test.

Reply:
It seems to me more logically to include removal of this test to the PR which actually will remove `from_reader` from `Reader<&[u8]>`. For now we ensure some behavior via that test and before encoding will be reworked, it is better at least keep the behavior and verify it by tests.

Also, after removal of that test the comment for a `bom` test function become outdated

Reply:
I'm just trying to make that PR smaller and more approachable - both of these patches are from there so if you'd prefer to keep it in that PR then I'll just close this one entirely.

Reply:
Yes, I think then it is better to include them in that PR. I, too, struggled with the temptation to split  my next PR into pieces, but it seems to me that it will be better if they are all together, although there are 20 commits :)